cistem/protocols/protocol_tomo_resample.py

```diff
@@ -0,0 +1,184 @@
+# * (at your option) any later version.
+# *
+# * This program is distributed in the hope that it will be useful,
+# * but WITHOUT ANY WARRANTY; without even the implied warranty of
+# * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# * GNU General Public License for more details.
+# *
+# * You should have received a copy of the GNU General Public License
+# * along with this program; if not, write to the Free Software
+# * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+# * 02111-1307  USA
+# *
+# *  All comments concerning this program package may be sent to the
+# *  e-mail address 'scipion@cnb.csic.es'
+# *
+# **************************************************************************
+
+
+'''
+A protocol to resample tomograms by Fourier cropping/padding using cisTEM.
+'''
+from cistem import Plugin
+from pwem.protocols import EMProtocol
+from pyworkflow import BETA
+from pyworkflow.protocol import PointerParam, IntParam
+from pyworkflow.utils import *
+from tomo.objects import SetOfTomograms, Tomogram
+from pwem.convert.headers import Ccp4Header
+
+OUTPUT_TOMO_NAME = 'resampledTomos'
+OUTPUT_DIR = 'extra/'
+
+class ProtTomoResample(EMProtocol):
+    '''
+    Resample tomograms by Fourier cropping/padding using cisTEM. This is equivalent to binning/unbinning operations but free of aliasing artifacts.
+
+    More info:
+        https://cistem.org
+    '''
+
+    _label = 'resample tomogram'
+    _possibleOutputs = {OUTPUT_TOMO_NAME: SetOfTomograms}
+    _devStatus = BETA
+
+    tomoList = []
+
+    def __init__(self, **args):
+        EMProtocol.__init__(self, **args)
+
+    # -------------------------- DEFINE param functions ----------------------
+    def _defineParams(self, form):
+        ''' Define the input parameters that will be used.
+        Params:
+            form: this is the form to be populated with sections and params.
+        '''
+        #     Example 'resample' input:
+        # 
+        #     resample
+        #  
+        #  
+        #         **   Welcome to Resample   **
+        #  
+        #              Version : 1.00
+        #             Compiled : Dec  2 2017
+        #                 Mode : Interactive
+        #  
+        # Input image file name [tomo2_L1G1_72_d3.rec]       : tomo2_L1G1-dose_filt.rec
+        # Output image file name [tomo2_L1G1_72_d3-bin8.rec] : tomo2_L1G1_72_d3-bin8.rec
+        # Is the input a volume [YES]                        : YES
+        # New X-Size [464]                                   : 464
+        # New Y-Size [464]                                   : 464
+        # New Z-Size [232]                                   : 232
+
+
+        # You need a params to belong to a section:
+        form.addSection(label=Message.LABEL_INPUT)
+        form.addParam('inTomograms', PointerParam,
+                        pointerClass='SetOfTomograms',
+                        allowsNull=False,
```

Comment:
allowsNull=False is not required for PointerParam type, base protocol validation checks this automatically